test/cpp_extensions/doubler.h

```diff
@@ -2,7 +2,7 @@
 
 struct Doubler {
   Doubler(int A, int B) {
-     tensor_ = at::CPU(at::kDouble).ones({A, B});
+    tensor_ = torch::as_variable(at::CPU(at::kDouble).ones({A, B}));
```

Comment:
Suggestion: add `torch::CPU(...)`, `torch::CUDA(...)`, and `torch::getType(...)` which will return `VariableType` instances and allocate variables.  Now we do not need torch::as_variable in common usage, while it still allows us to unpack the tensors in our own internal code.

Reply:
I like that!